streaming/src/main/scala/org/apache/spark/streaming/receiver/Receiver.scala

```diff
@@ -107,9 +107,6 @@ abstract class Receiver[T](val storageLevel: StorageLevel) extends Serializable
    */
   def onStop()
 
-  /** Override this to specify a preferred location (hostname). */
-  def preferredLocation : Option[String] = None
```

Comment:
So with this change, custom receiver implementations will need to re-implement this code? I am not sure that is a good idea, since this is part of a stable interface. For one, that breaks custom code and the other aspect is that it sets a bad precedent.